BitwardenShared/UI/Vault/VaultItem/CipherItemState.swift

```diff
@@ -140,6 +143,26 @@ struct CipherItemState: Equatable {
         }
     }
 
+    /// Whether or not this item can be deleted by the user.
+    /// New permission model from PM-18091
+    var canBeDeletedPermission: Bool {
+        // backwards compatibility for old server versions
+        guard let cipherPermissions = cipher.permissions else {
+            return canBeDeleted
+        }
+        return cipherPermissions.delete
+    }
+
+    /// Whether or not this item can be restored by the user.
+    /// New permission model from PM-18091
+    var canBeRestoredPermission: Bool {
+        // backwards compatibility for old server versions
+        guard let cipherPermissions = cipher.permissions else {
+            return isSoftDeleted
+        }
+        return cipherPermissions.restore && isSoftDeleted
+    }
+
```

Comment:
🎨 You can also refactor this a bit so you have less logic in the view and it's easier to remove the feature flag in the future. Additionally, it's safer to use this property as it's safe-guarded in-situ with the flag so the consumer doesn't need to know the existence of the flag:
```suggestion
    /// Whether or not this item can be restored by the user.
    /// New permission model from PM-18091
    var canBeRestoredPermission: Bool {
        // backwards compatibility for old server versions
        guard restrictItemDeletionFlagEnabled, let cipherPermissions = cipher.permissions else {
            return isSoftDeleted
        }
        return cipherPermissions.restore && isSoftDeleted
    }

```

Reply:
This one I can't add here and need to leave it in the view, because otherwise in order to maintain state, I would have to pass the Flag value to the init of the `CipherItemState` and I wan't to avoid that.

Reply:
Alright no problem